plugins/CoreAdminHome/OptOutManager.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\CoreAdminHome;
+
+use Piwik\Common;
+use Piwik\Nonce;
+use Piwik\Plugins\LanguagesManager\API as APILanguagesManager;
+use Piwik\Plugins\LanguagesManager\LanguagesManager;
+use Piwik\Plugins\PrivacyManager\DoNotTrackHeaderChecker;
+use Piwik\Tracker\IgnoreCookie;
+use Piwik\Url;
+use Piwik\View;
+
+class OptOutManager
+{
+    /** @var DoNotTrackHeaderChecker */
+    private $doNotTrackHeaderChecker;
+
+    /** @var array */
+    private $javascripts;
+
+    /** @var array */
+    private $stylesheets;
+
+    /** @var string */
+    private $title;
+
+    /**
+     * @param DoNotTrackHeaderChecker $doNotTrackHeaderChecker
+     */
+    public function __construct(DoNotTrackHeaderChecker $doNotTrackHeaderChecker = null)
+    {
+        $this->doNotTrackHeaderChecker = $doNotTrackHeaderChecker ?: new DoNotTrackHeaderChecker();
+
+        $this->javascripts = array(
+            'inline' => array(),
+            'external' => array(),
+        );
+
+        $this->stylesheets = array(
+            'inline' => array(),
+            'external' => array(),
+        );
+    }
+
+    /**
+     * @param string $javascript
+     * @param bool $inline
+     */
+    public function addJavascript($javascript, $inline = true)
+    {
+        $type = $inline ? 'inline' : 'external';
```

Comment:
Yes i think so. For example the CustomOptOut Plugin stores the css code in the database for each site.